# HG changeset patch
# User Rik <rik@octave.org>
# Date 1492558538 25200
#      Tue Apr 18 16:35:38 2017 -0700
# Node ID 7b95435e96a403c2b50547143c28b172885f62ba
# Parent  98bf881fafd1993f8456009cbaa36fe2ea577bb5
Reverse orientation of vertical uicontrol "sliders" for Matlab compatibility (bug #50818).

* SliderControl.cc (SliderControl::SliderControl): Call setInvertedAppearance
on slider for vertical sliders.

diff --git a/libgui/graphics/SliderControl.cc b/libgui/graphics/SliderControl.cc
--- a/libgui/graphics/SliderControl.cc
+++ b/libgui/graphics/SliderControl.cc
@@ -54,17 +54,20 @@ namespace QtHandles
   SliderControl::SliderControl (const graphics_object& go,
                                 QAbstractSlider* slider)
     : BaseControl (go, slider), m_blockUpdates (false)
   {
     uicontrol::properties& up = properties<uicontrol> ();
 
     slider->setTracking (false);
     Matrix bb = up.get_boundingbox ();
-    slider->setOrientation (bb(2) > bb(3) ? Qt::Horizontal : Qt::Vertical);
+    bool vertical_slider = ( bb(2) < bb(3) );
+    slider->setOrientation (vertical_slider ? Qt::Vertical : Qt::Horizontal);
+    if (vertical_slider)
+      slider->setInvertedAppearance (true);  // Matlab compatibility
     Matrix steps = up.get_sliderstep ().matrix_value ();
     slider->setMinimum (0);
     slider->setMaximum (RANGE_INT_MAX);
     slider->setSingleStep (octave::math::round (steps(0) * RANGE_INT_MAX));
     slider->setPageStep (octave::math::round (steps(1) * RANGE_INT_MAX));
     Matrix value = up.get_value ().matrix_value ();
     if (value.numel () > 0)
       {
